### PR TITLE
Updates on unfolding

### PR DESCRIPTION
**Pull request by davidwalter2**
This is a part of PR #445 that should not change the results
- Adding/updating a few plotting scripts
- Simplification/ clean up of code
- Adding some functionality in preparation for linearized unfolding
- The most notable change for users is that the genAxes argument in the setupCombine.py is now a list of dashed separated strings for the axes names, similar to the fit axes, in order to make a card from different input files with different gen axes

### CODE REVIEW

File: utilities/common.py [hunk 1]
```diff
@@ -238,11 +238,11 @@ def set_subparsers(subparser, name, analysis_label):
         if analysis_label not in axmap:
             raise ValueError(f"Unknown analysis {analysis_label}!")
         subparser.add_argument("--genAxes", type=str, nargs="+", 
-                               default=axmap[analysis_label], choices=["qGen", "ptGen", "absEtaGen", "ptVGen", "absYVGen"],
+                               default=axmap[analysis_label], choices=["qGen", "ptGen", "absEtaGen", "ptVGen", "absYVGen", "helicitySig"],
```
Comment (cippy) [inline utilities/common.py:241]: Are you sure the helicitySig axes has to go here? It was supposed to be added elsewhere, for technical reasons which right now I don't remember exactly. Could it be why you saw large difference for the theory agnostic fit ?

Comment (davidwalter2) [reply utilities/common.py:241]: For now I only added it in "choices" in preparation for future updates, I don't think it's causing the problem

**Comment by davidwalter2** [comment]
This would be ready for review/merge from my side
Here is the PR validation from this PR:
https://cmsmwbot.web.cern.ch/WMassAnalysis/PRValidation/PR493/2024_07_09/
Compared to the reference run:
https://cmsmwbot.web.cern.ch/WMassAnalysis/PRValidation/ReferenceRuns/2024_07_09_bb40c91/

**Comment by cippy** [comment]
It's not worth running CI for it, but if genes now takes a dash separated list, it might be worth specifying it in the help message (or would it also work by passing a normal list using the + argument? If not, maybe better to let the user know since it is not obvious how it should work otherwise)

**Comment by davidwalter2** [comment]
> It's not worth running CI for it, but if genAxes now takes a dash separated list, it might be worth specifying it in the help message (or would it also work by passing a normal list using the + argument? If not, maybe better to let the user know since it is not obvious how it should work otherwise)

It still takes a list because if you specify multiple input files it takes a dash separated string for each input file, i.e. a list of dash separated string (only supported for --hdf5). I can update the help message in one of the upcoming updates but if there is no other comment I would leave it for the moment

**Pull request merged**